src/main/java/core/basesyntax/service/RegistrationServiceImpl.java

```
@@ -3,12 +3,68 @@
 import core.basesyntax.dao.StorageDao;
 import core.basesyntax.dao.StorageDaoImpl;
 import core.basesyntax.model.User;
+import core.basesyntax.throwable.InvalidDataEntry;
 
 public class RegistrationServiceImpl implements RegistrationService {
+
+    static final int MIN_LOGIN_LENGTH = 6;
+    static final int MIN_PASSWORD_LENGTH = 6;
+    static final int MIN_AGE = 18;
     private final StorageDao storageDao = new StorageDaoImpl();
 
     @Override
     public User register(User user) {
-        return null;
+        if (user == null) {
+            throw new InvalidDataEntry("User is null.");
+        }
+        userExistCheck(user);
+        loginSizeCheck(user);
+        ageCheck(user);
+        passwordSizeCheck(user);
+        return storageDao.add(user);
+    }
+
+    @Override
+    public boolean userExistCheck(User user) {
+        if (user.getLogin() == null) {
+            throw new InvalidDataEntry("Login can't be null");
+        }
+        if (storageDao.get(user.getLogin()) != null) {
+            throw new InvalidDataEntry("User with such login already exists.");
+        }
+        return true;
+    }
+
+    @Override
+    public boolean loginSizeCheck(User user) {
+        if (user.getLogin() == null) {
+            throw new InvalidDataEntry("Login can't be null");
+        }
+        if (user.getLogin().length() < MIN_LOGIN_LENGTH) {
+            throw new InvalidDataEntry("Login must have at least 6 characters.");
```

This violates the checklist item: "Move all hardcoded values to [constant fields] and give them informative names." Replace the literal `6` in the login error message with the constant `MIN_LOGIN_LENGTH` (or construct the message using that constant) so messages do not hardcode magic numbers. Example change: `"Login must have at least " + MIN_LOGIN_LENGTH + " characters."`. See login size check.